src/logger/src/lib.rs

```diff
@@ -729,28 +729,56 @@ mod tests {
         error!("error");
 
         // Assert that initialization works only once.
+
         let log_file_temp =
-            NamedTempFile::new().expect("Failed to create temporary output logging file.");
-        let log_file = String::from(log_file_temp.path().to_path_buf().to_str().unwrap());
+            TempFile::new().expect("Failed to create temporary output logging file.");
+        let log_file = String::from(log_file_temp.as_path().to_path_buf().to_str().unwrap());
         let metrics_file_temp =
-            NamedTempFile::new().expect("Failed to create temporary metrics logging file.");
+            TempFile::new().expect("Failed to create temporary metrics logging file.");
         l.set_instance_id(TEST_INSTANCE_ID.to_string());
         assert!(l
             .init(
                 &app_info,
-                Box::new(log_file_temp),
-                Box::new(metrics_file_temp),
+                Box::new(
```

Comment:
Actually, because it might spread to another week, what do you think if we merge this like it is, if we get the approves? We can update with your suggestion in a subsequent PR. Thinking about it, this PR should've not been blocked at all. The updates could've been added in subsequent PRs (TempFile API changes, signal API refactoring). Please let me know what you think. I just want to not get blocked here everytime we find things that miss or need changes in `rust-vmm/vmm-sys-util`.

Reply:
I agree